src/app/workflow-app/workflow-app.component.ts

```diff
@@ -107,6 +112,31 @@ export class WorkflowAppComponent implements AfterContentInit {
     }
   }
 
+  @HostListener('document:mousedown', ['$event'])
```

Comment:
Couldn't we check whether zosmfServerConfigComponent has unsaved changes right in the window close handler?

Reply:
Right. I thought about that today too, and am going to change it to this, avoiding the reference of class names in the CSS.

Reply:
Hey Leonty, 
The problem is, for some reason the closeHandler executes, even if we are just setting the handler itself, not closing the program. It also doesn't activate on close() by default for some reason as well. I'll message you separately if poking around for an hour or two doesn't help, but it looks a little more complicated than assuming that the logic can be handled only within the closeHandler, especially if the closeHandler seems to not exactly do what it's supposed to.
@lchudinov